### DISCUSSION TITLE
Scrollbar styling support

### DISCUSSION THREAD

**Discussion by mathurinfarms** [tailwindlabs/tailwindcss#9915] [Ideas]
# This is not an issue - more of a recommendation

By default, TailwindCSS doesn't allow for custom scrollbars / styling.

There's currently a plugin on npm called [tailwind-scrollbar](https://www.npmjs.com/package/tailwind-scrollbar)

Even Tailwind's website uses custom scrollbars in the documentation - it would be great to add scrollbar styling natively instead of having to use plugins or adding custom CSS:

![image](https://user-images.githubusercontent.com/97749828/203683318-dd961a46-6fdb-4db6-b345-c7841bff8efe.png)

**Comment by agroot2002** [comment]
I don't want to use plugin for this , can you tell me how I can add scrollbar property to my tailwind project . More specifically I want to hide my scrollbar but I am unable to find a solution .

**Reply by agroot2002** [reply]
Hey I got the answer , I reverse engineered the code of the repo you provided . So the solution I applied is , I created a plugin in **`tailwind.config.js`** like this :

```
  const plugin = require('tailwindcss/plugin')
  
  plugins: [
    plugin(function ({ addUtilities }) {
      addUtilities({
        '.scrollbar-none': {
          '&::-webkit-scrollbar': {
            'display': 'none'
          }
        }
      })
    })
  ],
  
```

now as we add the class 'scrollbar-none' to the component after creating the plugin . We can see the results !

**Reply by mathurinfarms** [reply]
Ah that's a good idea actually. We could also style the scrollbar this way. Thanks!

**Reply by lukewarlow** [reply]
This should have a `scrollbar-width: none` for Firefox support.

```
'.scrollbar-none': {
        'scrollbar-width': 'none',
        '&::-webkit-scrollbar': {
          display: 'none',
        },
}
 ```

**Comment by lukewarlow** [comment]
See https://github.com/lukewarlow/tailwind-scrollbar-utilities for an alternative plugin that's focussed on the standard CSS properties scrollbar-color and scrollbar-width. (With a fallback to ensure hiding scrollbars works in all browsers)

These properties are supported inside Firefox and Chrome (from 121).

**Comment by xegulon** [comment]
What about having tailwind classes for scrollbar color, scrollbar bg color and width?

@adamwathan would you accept a PR in this direction?

**Reply by lukewarlow** [reply]
I already put together a PR for scrollbar-width and scrollbar-gutter and they were rejected at the time fwiw.

**Reply by xegulon** [reply]
@adamwathan does tailwind not intend to support this specific feature? if yes, why so?

**Reply by mathurinfarms** [reply]
@adamwathan It would be great to see such a feature indeed. That's the whole reason I suggested this about a year ago

**Comment by merobal** [comment]
@adamwathan ping on this. It became official even in Chrome: https://developer.chrome.com/docs/css-ui/scrollbar-styling

**Comment by MrOxMasTer** [comment]
The discussion has been going on for 2 years, 150k people download the plugin, but they still haven't added it. Sadly
![image](https://github.com/tailwindlabs/tailwindcss/assets/59291123/117d99a1-be9b-440b-8795-ac208bbd8a7e)

**Reply by thiagovianav** [reply]
3 years and  a half now lol
